go/storage/mkvs/db/badger/badger.go

```diff
@@ -891,6 +1056,11 @@ func (s *badgerSubtree) PutNode(depth node.Depth, ptr *node.Pointer) error {
 	if err = s.batch.bat.Set(nodeKeyFmt.Encode(&h), data); err != nil {
 		return err
 	}
+	if s.batch.multipartNodes != nil {
+		if err = s.batch.multipartNodes.Set(multipartRestoreNodeLogKeyFmt.Encode(&h), []byte{}); err != nil {
```

Comment:
As mentioned before, this can remove pre-existing nodes (e.g., if `h` existed before, this is not checked anywhere). You should only add a node to `multipartRestoreNodeLogKeyFmt` in case it didn't exist before. Also add a test for this to make sure (e.g., create a root at version 1, then restore a root at version 2 which shares some nodes with the root at version 1, but abort the multipart restore instead of finalizing it, then iterate over root at version 1 to make sure everything is still there).